eos/b-decays/b-to-gamma-l-nu.cc

```diff
@@ -0,0 +1,214 @@
+
+        static const constexpr double e_l = -1.0;
+
+        static const std::vector<OptionSpecification> options;
+
+        Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+            model(Model::make(o.get("model", "SM"), p, o)),
+            form_factors(FormFactorFactory<PToGamma>::create("B->gamma::" + o.get("form-factors", "FLvD2022QCDF"), p, o)),
+            alpha_qed(p["QED::alpha_e(m_b)"] ,u),
+            g_fermi(p["WET::G_Fermi"], u),
+            v_ub_abs(p["CKM::abs(V_ub)"], u),
+            hbar(p["QM::hbar"], u),
+            m_B(p["mass::B_u"], u),
+            f_B(p["decay-constant::B_u"], u),
+            tau_B(p["life_time::B_u"], u)
+        {
+            u.uses(*model);
+        }
+
+        double decay_width(const double & E_gamma_min) const
+        {
+            // cf. [BBJW:2018A], eq. (2.6)
+            const double prefactor = alpha_qed * power_of<2>(g_fermi * v_ub_abs) / (6.0 * power_of<2>(M_PI));
+
+            auto integrand = [&](const double & E_gamma) -> double {
+                const double F_V = form_factors->F_V(E_gamma);
+                const double F_A = form_factors->F_A(E_gamma);
+                return m_B * power_of<3>(E_gamma) * (1.0 - 2.0 * E_gamma / m_B)
+                    * ( power_of<2>(std::abs(F_V)) + power_of<2>( F_A + e_l * f_B / E_gamma ) );
+            };
+
+            return prefactor * integrate<GSL::QAGS>(integrand, E_gamma_min, m_B / 2.0);
+        }
+
+        double branching_ratio(const double & E_gamma_min) const
+        {
+            return decay_width(E_gamma_min) * tau_B / hbar;
+        }
+
+        double diff_decay_width(const double & E_gamma, const double & costheta) const
+        {
+            const double E_ell = (-2.0 * E_gamma * m_B + power_of<2>(m_B)) / (2.0 * (-E_gamma + costheta * E_gamma + m_B));
+            const double dE_ell_dcostheta = (E_gamma * (2.0 * E_gamma - m_B) * m_B / power_of<2>((-1 + costheta) * E_gamma + m_B)) / 2.0;
+
+            const double x_gamma = 2.0 * E_gamma / m_B;
+            const double x_ell   = 2.0 * E_ell   / m_B;
+            const double x_nu    = 2.0 * (1.0 - (E_gamma + E_ell) / m_B);
+
+            const double F_V = form_factors->F_V(E_gamma);
+            const double F_A = form_factors->F_A(E_gamma) + e_l * f_B / E_gamma; // mind different definitions of F_A between [BBJW:2018A] and [BR:2011A]
+
+            // cf. [BR:2011A], eq. (2.7)
+            const double dGamma_dE_gamma_dE_ell =
+                alpha_qed * power_of<2>(g_fermi * v_ub_abs) / (16.0 * power_of<2>(M_PI))
+                * power_of<3>(m_B) * (1.0 - x_gamma)
+                * (
+                      power_of<2>(1.0 - x_nu ) * power_of<2>(F_A + F_V)
+                    + power_of<2>(1.0 - x_ell) * power_of<2>(F_A - F_V)
+                  );
+            return dGamma_dE_gamma_dE_ell * dE_ell_dcostheta;
+        }
+
+        std::tuple<double, double> Gamma_F_Gamma_B(const double & E_gamma_min) const
```

Comment:
In principle, this is a good idea. In general, it's beneficial to reduce the number of implementation of the dGamma/dEgamma, which you now have twice. I would probably only write one function, returning the difference of Gamma_F and Gamma_B. This implementation is sound, though. So I prefer to keep it as is. It might be optimized at a later stage (post merge), using the ``IntermediateResult`` framework